src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/maven/WriteMavenPomStep.java

```diff
@@ -41,26 +45,32 @@
 import org.kohsuke.stapler.DataBoundConstructor;
 import org.kohsuke.stapler.DataBoundSetter;
 
+import javax.annotation.CheckForNull;
 import javax.annotation.Nonnull;
 import javax.inject.Inject;
 import java.io.FileNotFoundException;
 import java.io.OutputStream;
 import java.io.Serializable;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Set;
 
 /**
  * Writes a maven pom file to the current working directory.
- *
+ * The step must be invoked within the {@link com.cloudbees.groovy.cps.NonCPS} context.
```

Comment:
I do not really see what that has to do with anything. `Model` is `Serializable`, and `program.dat` serialization is not subject to JEP-200. However you run the step, `ArgumentsAction` will be saved. If that breaks today,

* That is a problem for `workflow-api` to fix, which I think it already did.
* It is because this step should not exist. Deprecate it.